app/models/claims/training_allowance.rb

```diff
@@ -35,8 +35,8 @@ def available?
                    :REFRESHER_TRAINING_HOURS
 
   def mentor_training_scope
-    @mentor_training_scope ||= provider.mentor_trainings
-                                       .where(mentor:)
+    @mentor_training_scope ||= mentor.mentor_trainings
+                                       .where(provider:)
                                        .joins(claim: { claim_window: :academic_year })
```

Comment:
Doesn't this produce the exact same query? 🤔 Any reason why it changed in this PR?

```ruby
provider.mentor_trainings.where(mentor:)
# vs
mentor.mentor_trainings.where(provider:)
```

No strong preference as to which is better. If you prefer the newer version, then we can keep it! 🙂

Reply:
There's an error raised when starting from `provider` due to the STI (If I recall correctly) but this is mitigated by using `mentor`, seemed the easiest solution for the issue

Reply:
🤨 How odd. I've just dug into it and it looks like a couple of misconfigured factories.

I'll push up a commit to fix them, and then tests pass with the code remaining as before.

Reply:
Resolved by d033117cb2b5a93e44386e2e8694c24468f1a6bd